enhancements/baremetal/baremetal-provisioning-config.md

```diff
@@ -0,0 +1,188 @@
+these configurations items.
+
+### Proposal
+
+Before BareMetal Hosts can be matched to Machines, they need to be connected to their
+provisioning network for them to be PXE booted and given an IP address. To make
+this happen, the provisioning service needs to know which NIC, provisioning network,
+IP address and image URLs to use to download and boot images on these servers.
+
+A new CR called "Metal3ProvisioningController" or just "Metal3Controller" would be
+created within the operator scope.
+
+This new CR would consist of the following:
+
+1. ProvisioningInterface : This is the interface name on the underlying baremetal
+server which would be physically connected to the provisioning network. This
+configuration is needed only for the underlying provisioning service (Ironic)
+and could have values like "eth1" or "ens3".
+
+2. ProvisioningIP : This is the IP address used to to bring up a NIC on the
+baremetal server for provisioning. This is also a value that is useful just to the
+provisioning service. This value should not be in the DHCP range and should not
+be used in the provisioning network for any other purpose (should be a free IP
+address.) It is expected to be provided in the format : 10.1.0.3.
+
+3. ProvisioningNetworkCIDR : This is the provisioning network with its CIDR. The
+ProvisioningIP and the ProvisioningDHCPRange are expected to be within this network.
+The value for this config is expected to be provided in the format : 10.1.0.0/24.
+
+4. ProvisioningDHCPRange : This is a range of IP addresses from which baremetal
+hosts can be assigned their IP address. This range should be provided as two
+comma seperated IP addresses where the addresses represent a contiguous range of
+IPs that are available to be allocated. It is expected to be provided in the format :
+10.1.0.10, 10.1.0.100. This is an optional parameter and if the range specified is
+empty or null, no DHCP services will be started. This option to opt out is helpful
+in environments where external DHCP services are provided for the provisioning network.
+
+### User Stories [optional]
+
+This enhancement will allow users to have their Barametal IPI deployments
+customizable to their specific hardware and network requirements.
+
+This also allows Baremetal IPI deployments to take place without manual
+workarounds like creating a ConfigMap for the config (which is the current
+approach being used in 4.2 and 4.3.) 
+
+The new config items would be set by the installer and will be used by the
+machine-api-operator (MAO) to generate more config items that are derivable from
+these basic parameters. Put together, these config parameters are passed in as
+environment variables to the various containers that make up a metal3 baremetal
+IPI deployment.
+
+This baremetal provisioning CR contains configueation data for the provisioning
+services, which are not values that should be configured by the end user via
+BareMetalHost objects.
+
+## Design Details
```

Comment:
a CRD requires at least..
`name` 
`apigroup`
`version`

so let's make sure we have clearly specify theses in the design.

And for the controller - CR relation we would need to define the `namespace` and the `name` if it's a singleton. (if this singleton is for the cluster maybe the name should be `cluster` like others..)

Reply:
@sadasu see https://github.com/openshift/openshift-docs/blob/c5b7bdc145d8f79a60062934375117f5a22fa5ca/installing/install_config/customizations.adoc#operator-configuration-resources

Reply:
> (if this singleton is for the cluster maybe the name should be `cluster` like others..)

@abhinavdahiya would naming the singleton `cluster`, imply cluster scope?

Reply:
Updated the instance name to main/default.